source/adapters/cuda/usm.cpp

```diff
@@ -396,40 +465,58 @@ ur_usm_pool_handle_t_::ur_usm_pool_handle_t_(ur_context_handle_t Context,
     pNext = BaseDesc->pNext;
   }
 
-  auto MemProvider =
-      umf::memoryProviderMakeUnique<USMHostMemoryProvider>(Context, nullptr)
-          .second;
-
-  HostMemPool =
-      umf::poolMakeUnique<usm::DisjointPool, 1>(
-          {std::move(MemProvider)},
-          this->DisjointPoolConfigs.Configs[usm::DisjointPoolMemType::Host])
-          .second;
+  ur_result_t Ret;
+  std::tie(Ret, PoolManager) =
+      usm::pool_manager<usm::pool_descriptor>::create();
+  if (Ret) {
+    throw UsmAllocationException(Ret);
+  }
 
   auto Device = Context->DeviceID;
-  MemProvider =
-      umf::memoryProviderMakeUnique<USMDeviceMemoryProvider>(Context, Device)
-          .second;
-  DeviceMemPool =
-      umf::poolMakeUnique<usm::DisjointPool, 1>(
-          {std::move(MemProvider)},
-          this->DisjointPoolConfigs.Configs[usm::DisjointPoolMemType::Device])
-          .second;
-
-  MemProvider =
-      umf::memoryProviderMakeUnique<USMSharedMemoryProvider>(Context, Device)
-          .second;
-  SharedMemPool =
-      umf::poolMakeUnique<usm::DisjointPool, 1>(
-          {std::move(MemProvider)},
-          this->DisjointPoolConfigs.Configs[usm::DisjointPoolMemType::Shared])
-          .second;
+  auto UrUSMPool = reinterpret_cast<ur_usm_pool_handle_t>(this);
+
+  // TODO: Replace this with appropriate usm::pool_descriptor 'create' static
```

Comment:
I think we can do this already, we've got this https://github.com/oneapi-src/unified-runtime/blob/39074a8eb604cc1a74180002719dadb3699fd778/source/common/ur_pool_manager.hpp#L176

Reply:
Unfortunately, `pool_descriptor::create` creates an unnecessary descriptor for `SharedReadOnly` memory type, which is not used by CUDA adapter.

Reply:
oh right, of course